File: codeforces/Abby-2013/A/a.cpp
#include<iostream>
#include<locale>
#include<vector>
using namespace std;

int main()
{
  ios::sync_with_stdio(false);

  string s; cin >> s;

  int als = 0;
  vector<bool> used(30, false);

  long long sum = 1;

  for(int i = 0; i < s.size(); i++)
  {
    int multiplier = 1;

    if(s[i] == '?')
    {
      if(i > 0)
      {
        multiplier = 10;
      }
      else
      {
        multiplier = 9;
      }
    }
    else if(isalpha(s[i]))
    {
      if(used[s[i] - 'A'])
      {
        multiplier = 1;
      }
      else if(i > 0)
      {
        multiplier = 10 - als;
      }
      else
      {
        multiplier = 9;
      }
    }

    sum *= multiplier;

    if(isalpha(s[i]) && !used[s[i] - 'A'])
    {
      als++;
      used[s[i] - 'A'] = true;
    }
  }

  cout << sum << endl;

  return 0;
}

File: codeforces/Abby-2013/A/a.rb
s=gets

used = {}
als = 0
sum = 1

class String
  def is_number?
    true if Float(self) rescue false
  end
end

s.each_char.each_with_index do |c, i|
  mult = 1

  if c == '?'
    if i > 0
      mult = 10
    else
      mult = 9
    end
  elsif c.ord >= 'A'.ord && c.ord <= 'Z'.ord
    if used[c]
      mult = 1
    elsif i > 0
      mult = 10 - als
    else
      mult = 9
    end
  end

  sum *= mult

  if c.ord >= 'A'.ord && c.ord <= 'Z'.ord && !used[c]
    als += 1
    used[c] = true;
  end
end

puts sum
